src/main/java/gregtech/api/mui/widget/appeng/fluid/AEFluidConfigSlot.java

```diff
@@ -0,0 +1,116 @@
+package gregtech.api.mui.widget.appeng.fluid;
+
+import gregtech.api.mui.GTGuiTextures;
+import gregtech.api.mui.sync.appeng.AEFluidSyncHandler;
+import gregtech.api.mui.widget.appeng.AEConfigSlot;
+import gregtech.api.mui.widget.appeng.AEStackPreviewWidget;
+import gregtech.api.util.KeyUtil;
+import gregtech.api.util.TextFormattingUtil;
+import gregtech.client.utils.RenderUtil;
+import gregtech.common.metatileentities.multi.multiblockpart.appeng.stack.WrappedFluidStack;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidUtil;
+
+import appeng.api.storage.data.IAEFluidStack;
+import com.cleanroommc.modularui.api.widget.Interactable;
+import com.cleanroommc.modularui.integration.jei.JeiGhostIngredientSlot;
+import com.cleanroommc.modularui.screen.RichTooltip;
+import com.cleanroommc.modularui.screen.viewport.ModularGuiContext;
+import com.cleanroommc.modularui.theme.WidgetTheme;
+import com.cleanroommc.modularui.value.sync.SyncHandler;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.BooleanSupplier;
+
+public class AEFluidConfigSlot extends AEConfigSlot<IAEFluidStack>
+                               implements Interactable, JeiGhostIngredientSlot<FluidStack> {
+
+    public AEFluidConfigSlot(boolean isStocking, int index, BooleanSupplier isAutoPull) {
+        super(isStocking, index, isAutoPull);
+        tooltipAutoUpdate(true);
+    }
+
+    @Override
+    public void onInit() {
+        super.onInit();
+        getContext().getJeiSettings().addJeiGhostIngredientSlot(this);
+    }
+
+    @Override
+    protected void buildTooltip(@NotNull RichTooltip tooltip) {
+        IAEFluidStack config = getSyncHandler().getConfig(index);
+        if (config == null) {
+            super.buildTooltip(tooltip);
+        } else {
+            KeyUtil.fluidInfo(((WrappedFluidStack) config).getDelegate(), tooltip, false, true, true);
+        }
+    }
+
+    @Override
+    public @NotNull AEFluidSyncHandler getSyncHandler() {
+        return (AEFluidSyncHandler) super.getSyncHandler();
+    }
+
+    @Override
+    public boolean isValidSyncHandler(SyncHandler syncHandler) {
+        return syncHandler instanceof AEFluidSyncHandler;
+    }
+
+    @Override
+    public void draw(ModularGuiContext context, WidgetTheme widgetTheme) {
+        IAEFluidStack config = getSyncHandler().getConfig(index);
+        if (config != null) {
+            FluidStack stack = ((WrappedFluidStack) config).getDelegate();
+            RenderUtil.drawFluidForGui(stack, stack.amount, 1, 1, 17, 17);
+
+            if (!isStocking) {
+                String amount = TextFormattingUtil.formatLongToCompactString(stack.amount, 4);
+                RenderUtil.renderTextFixedCorner(amount, 17d, 18d, 0xFFFFFF, true, 0.5f);
```

Comment:
This amount is not updated after scrolling the fluid to change the config amount. Nor after clicking the fluid and setting the amount in the text box.